proposals/lip-define_new_transaction_schema.md

```diff
@@ -0,0 +1,263 @@
+            "fieldNumber": 5
+        },
+        "params": {
+            "dataType": "bytes",
+            "fieldNumber": 6
+        },
+        "signatures": {
+            "dataType": "array",
+            "items": {
+                "dataType": "bytes",
+                "length": ED25519_SIGNATURE_LENGTH 
+            },
+            "fieldNumber": 7
+        }
+    }
+}
+```
+
+
+#### Validation
+
+For a transaction `trs` to be valid, it must satisfy the following:
+
+* `trs` must follow the `transactionSchema`. 
+* `trs.params` is of length less than or equal to `MAX_PARAMS_SIZE` .
+
+These checks are performed in the [static validation stage][lip-0055#block-processing] of the block processing. 
+
+
+### Serialization
+
+The serialization of an object of type `Transaction` is described in the following pseudocode.
+
+```python
+def encodeTransaction(trs: Transaction) -> bytes:
+    paramsSchema = JSON schema corresponding to (trs.module, trs.command) pair
+    trs.params = encode(paramsSchema,trs.params)
+    return encode(transactionSchema,trs)
+```
+
+### Deserialization
+
+Consider a binary message `trsMsg`, corresponding to a serialized transaction. The deserialization procedure is as follows:
+
+```python
+def decodeTransaction(trsMsg: bytes) -> Transaction: 
+    trsData = decode(transactionSchema,trsMsg)
+    paramsSchema = JSON schema corresponding to (trsData.module, trsData.command) pair
+    trsData.params = decode(paramsSchema,trsData.params)
+    return trsData
+```
+
+### Transaction Signature Calculation
+
+Consider a data structure `unsignedTrs` representing a valid `Transaction` object in which the signatures array is initialized to the default value (an empty array). The following function calculates a signature of the object on a certain chain with secret key `sk`. 
+
+```python
+def computeTransactionSignature(sk: PrivateKeyEd25519, unsignedTrs: Transaction, networkIdentifier: bytes) -> SignatureEd25519:
+    serializedTrs = encodeTransaction(unsignedTrs)
+    return signMessage(sk, MESSAGE_TAG_TRANSACTION, networkIdentifier, serializedTrs)
+```
+
+Here `networkIdentifier` is the [network identifier](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0037.md#network-identifiers) for the chain and the function `signMessage` is defined in [LIP 0037](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0037.md#signing-and-verifying-with-ed25519).
+
+### Transaction Signature Validation
+
+Signature validation is done using the `verifySignatures` function defined in [LIP 0041](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0041.md#transaction-verification). 
+
+
+
+## Backwards Compatibility
+
+This LIP results in a hard fork as nodes following the proposed protocol will reject transactions following the previous schema, and nodes following the previous protocol will reject transactions following the proposed schema.
+
+
+## Reference Implementation
+
+TBD
+
+
+## Appendix
```

Comment:
I would suggest to remove the appendix for now if it is outdated (and maybe copy it to a separate document).

Reply:
agree. I could also prepare it, otherwise

Reply:
I commented out the appendix for now.